secure/storage/vault/src/lib.rs

```diff
@@ -53,6 +53,42 @@ impl Client {
         Self { host, token }
     }
 
+    pub fn create_policy(&self, policy_name: &str, policy: &Policy) -> Result<(), Error> {
+        let response = ureq::post(&format!("{}/v1/sys/policy/{}", self.host, policy_name))
+            .set("X-Vault-Token", &self.token)
+            .timeout_connect(10_000)
+            .send_json(policy.to_json()?);
+        if response.ok() {
+            Ok(())
+        } else {
+            Err(response.into())
+        }
+    }
+
+    pub fn read_policy(&self, policy_name: &str) -> Result<Policy, Error> {
+        let response = ureq::get(&format!("{}/v1/sys/policy/{}", self.host, policy_name))
+            .set("X-Vault-Token", &self.token)
+            .timeout_connect(10_000)
+            .call();
+        match response.status() {
+            200 => Ok(Policy::from_json_str(&response.into_string()?)?),
+            _ => Err(response.into()),
+        }
+    }
+
+    pub fn create_token(&self, policies: Vec<&str>) -> Result<String, Error> {
+        let response = ureq::post(&format!("{}/v1/auth/token/create", self.host))
```

Comment:
what's the authentication of this API? root only?

Reply:
In general, the only path allowed by a token is its own cubby hole (think a private secret). It needs to be given permissions to do anything else. So without additional permissions, this would only be accessible by the root token.

Reply:
This is only used for tests right? In actual deployments the Vault permissions/token will be setup outside of Libra.

Reply:
Answering on github for transparency. Minimally yes. Though there may be value in having this client library be used by multiple tools so we can have consistency in how we interface with the various storage backends including Vault.